packages/store/addon/-private/system/store/record-data-store-wrapper.ts

```diff
@@ -13,13 +12,17 @@ type IdentifierCache = import('../../identifiers/cache').IdentifierCache;
 type RecordData = import('../../ts-interfaces/record-data').RecordData;
 type AttributesSchema = import('../../ts-interfaces/record-data-schemas').AttributesSchema;
 type RelationshipsSchema = import('../../ts-interfaces/record-data-schemas').RelationshipsSchema;
-
+type RelationshipDefinition = import('@ember-data/model/-private/system/relationships/relationship-meta').RelationshipDefinition;
 /**
   @module @ember-data/store
 */
 
 type StableIdentifierOrString = StableRecordIdentifier | string;
 
+function metaIsRelationshipDefinition(meta: unknown): meta is RelationshipDefinition {
```

Comment:
meta needs to be an object and not null for this to be safe to run

Reply:
we already check that it's not false-y prior to running this.